tests/test_main.py

```diff
@@ -46,6 +47,19 @@ def test_login_with_valid_id(monkeypatch, capsys, test_instructor):
 
     # Assert
     captured = capsys.readouterr()
+
+    # Determine expected semester and year
+    current_month = datetime.datetime.now().month
```

Comment:
In the test, mock datetime.now() to return some arbitrary date instead of using the current date and time.